scripts/mods/msu_persistent_data_manager.nut

```diff
@@ -0,0 +1,93 @@
+this.msu_persistent_data_manager <- {
+	m = {
+		ModConfigPath = "mod_config/"
+		Mods = {}
+	},
+	
+	function create()
+	{
+		this.importModConfigFiles()
+	}
+
+	function importModConfigFiles()
+	{
+		local persistentDirectory = this.IO.enumerateFiles(this.m.ModConfigPath);
+		if (persistentDirectory == null)
+		{
+			return
+		}
+
+		foreach(file in persistentDirectory)
+		{
+			local components = split(file, "/");
+			local modID = components[1];
+			local settingsType = components[2];
+			::printWarning(format("Checking file, potential modID: '%s' and settingstype '%s'.", modID, settingsType), this.MSU.MSUModName, "persistence");
+			this.setMod(modID);
+			if(!(settingsType in this.getMods()[modID]))
+			{
+				this.getMods()[modID][settingsType] <- file;
+			}
+		}
+	}
+	
+
+	function getMods()
+	{
+		return this.m.Mods
+	}
+
+	function getMod(_modID)
+	{
+		if(!(_modID in this.getMods()))
```

Comment:
Don't think this error checking is necessary, if you wanna add the logerror though, maybe throw after?

Reply:
After reading some more, maybe this function isn't well named? have a function called getMod and another call hasMod perhaps?

Reply:
Sure